Userland/Games/Chess/Engine.cpp

```diff
@@ -60,3 +65,33 @@ void Engine::handle_bestmove(Chess::UCI::BestMoveCommand const& command)
 
     m_bestmove_callback = nullptr;
 }
+
+void Engine::handle_unexpected_eof()
+{
+    VERIFY(m_pid != -1);
```

Comment:
I agree that in the vast majority of cases, we would expect the process to have already terminated at this point. I do this because the notifier receiving an `EOF` does not guarantee that the process has terminated. 

I've added a comment to explain the reasoning behind this. I'd also be happy to remove it as you suggest if you think that would be better, as it covers a bit of an edge case.

Reply:
> I do this because the notifier receiving an EOF does not guarantee that the process has terminated.

If that's true, it's a big no no for this PR. The engine quitting when the UCI stream reach EOF is the whole point of this bugfix.


If you're saying that it might have not terminated *yet*, that's not an issue. Currently, `Chess` is babysitting then engine, but it shouldn't be the case, let the engine take its own decisions.

Reply:
I've removed the additional code to kill the `ChessEngine` process. I don't think that allowing the engine process to persist after the GUI has exited is necessarily a good idea, but I can revisit this later if it actually becomes a problem.